### PR TITLE
boards/nucleo144-f412: initial support

### PR DESCRIPTION
**Pull request by aabadie**
This PR adds support for the [nucleo144-f412 board](http://www.st.com/content/st_com/en/products/evaluation-tools/product-evaluation-tools/mcu-eval-tools/stm32-mcu-eval-tools/stm32-mcu-nucleo/nucleo-f412zg.html).

~~Not fully tested yet.~~

### CODE REVIEW

File: boards/nucleo144-f412/Makefile.features [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_cpuid
+FEATURES_PROVIDED += periph_i2c
+FEATURES_PROVIDED += periph_gpio
+FEATURES_PROVIDED += periph_rtc
+FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_timer
+FEATURES_PROVIDED += periph_uart
+FEATURES_PROVIDED += periph_pwm
```
Comment (kYc0o) [inline boards/nucleo144-f412/Makefile.features]: Alphabetical order.

File: cpu/stm32f4/ldscripts/stm32f412zg.ld [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @author          Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author          Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+MEMORY
+{
+    rom (rx)    : ORIGIN = 0x08000000, LENGTH = 1024K
```
Comment (kYc0o) [inline cpu/stm32f4/ldscripts/stm32f412zg.ld]: I think here you can put `1M`, but thats optional.

**Comment by aabadie** [comment]
fixed the alphabetical order.

Thanks for the reviews @kYc0o :)

**Comment by vincent-d** [comment]
Needs rebase

**Comment by aabadie** [comment]
rebased

**Comment by aabadie** [comment]
ping @kYc0o. Are you ok with the changes here ?

**Comment by kYc0o** [comment]
Alles gut hier! Go!

**Pull request merged**